### PR TITLE
Add basic tests

### PR DESCRIPTION
**Pull request by alexanderGugel**
* Minor bug fixes needed to make tests pass
* Basic unit tests
* Accumulator now works with arrays of arbitrary length
* requiring DesktopEmulationMode still has side effects, but can
  be disabled
* Minor fixes in Matrix
* Minor fixes in Quaternion
* Minor fixes in Random
* Minor fixes in MultipleTransition
* Add tests

Related: #545 #79

### CODE REVIEW

**Comment by TheAlphaNerd** [comment]
oh my glob

File: package.json [hunk 1]
```diff
@@ -99,11 +100,16 @@
   },
   "homepage": "https://famo.us",
   "devDependencies": {
+    "cssify": "^0.6.0",
```
Comment (TheAlphaNerd) [inline package.json]: this doesn't need to be duplicated in both devDep's and Deps

File: package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
   ],
   "scripts": {
     "test": "npm run lint -s",
+    "tap-test": "prova -t provify ./tests/**/**/*.spec.js ./tests/**/**/**/*.spec.js ./tests/**/*.spec.js -b",
```
Comment (TheAlphaNerd) [inline package.json]: I have an implementation using testling that we might want to explore.  It is headless and has better stack traces

**Comment by alexanderGugel** [comment]
Removed Prova, please don't merge yet - I'm factoring out the bug fixes into separate PRs.

**Comment by alexanderGugel** [comment]
Before merging this in, the following PRs are required (otherwise the tests would fail): 

Accumulator fix: #570
Matrix fix: #536
DesktopEmulationMode rewrite (no longer breaking): #555
Quaternion fix: #590
MultipleTransition fix:  #594

**Comment by TheAlphaNerd** [comment]
I'm not seeing the update to prova in the package.json... have you force pushed to updates?

**Comment by alexanderGugel** [comment]
Sorry, one second.

**Comment by alexanderGugel** [comment]
Closing because of #595

**Pull request closed without merging**